### PR TITLE
DOC: Fix formatting; add section titles; add more description.

### PR DESCRIPTION
**Pull request by danielballan**
- code-block needs a blank line before the code begins or it does
  not render as code
- I added some new section headings to make it easier to quickly find things.
- I moved a paragraph that had gotten pushed too far down the page.
- I added some explanation around handlers, with a link to the new assets
  section.

### CODE REVIEW

File: doc/source/configuration.rst [hunk 1]
```diff
@@ -74,21 +83,21 @@ classes that load externally stored data.
...
 
+Coping with Moved Files or Different Mount Points
+-------------------------------------------------
+
 Optionally, you may set a root_map, which comes in handy when the handlers
 involves mounted files that have been moved to a different mount point in the
 file system.
+
 .. code-block:: yaml
+
```
Comment (danielballan) [inline doc/source/configuration.rst]: (This blank line is the important one, FYI.)

File: doc/source/configuration.rst [hunk 2]
```diff
@@ -74,21 +83,21 @@ classes that load externally stored data.
...
     root_map : {'old_root': 'new_root',
                 'old_root2' : 'new_root2',}
 
 where ``old_root`` and ``old_root2`` are the old mount points and ``new_root``
 and ``new_root2`` their respective new mount points.
 
-In these examples, the classes used happen to be from the ``databroker``
```
Comment (danielballan) [inline doc/source/configuration.rst:87]: This was just moved (back) up closer to the relevant section.

File: doc/source/configuration.rst [hunk 3]
```diff
@@ -74,21 +83,22 @@ classes that load externally stored data.
...
 Optionally, you may set a root_map, which comes in handy when the handlers
 involves mounted files that have been moved to a different mount point in the
 file system.
+
 .. code-block:: yaml
-    root_map : {'old_root': 'new_root',
-                'old_root2' : 'new_root2',}
+
+    root_map:
+        old_root: new_root
```
Comment (jrmlhermitte) [inline doc/source/configuration.rst:96]: eek, thanks for this fix, my fault!

File: doc/source/configuration.rst [hunk 4]
```diff
@@ -74,21 +83,22 @@ classes that load externally stored data.
             module: 'databroker.assets.path_only_handlers'
             class: 'RawHandler'
 
+Coping with Moved Files or Different Mount Points
```
Comment (jrmlhermitte) [inline doc/source/configuration.rst:86]: :+1:

**Pull request merged**